### PR TITLE
Adding additional QueueItem types and handlers

### PR DESCRIPTION
**Pull request by quilime**
Adding a Midi Note QueueItem and enabling the task queue to handle and send MIDI notes.

### CODE REVIEW

File: src/PureDataNode.h [hunk 1]
```diff
@@ -88,6 +94,7 @@ class PureDataNode : public ci::audio::Node, public pd::PdReceiver {
   void sendList(const std::string &dest, const pd::List &list);
   void sendMessage(const std::string &dest, const std::string &msg,
                    const pd::List &list = pd::List());
+  void sendNoteOn(int port, int pitch, int velocity);
```
Comment (notlion) [inline src/PureDataNode.h]: Curious why `port` is used here and `channel` in the `NoteOn` struct. Seems like we should pick one or the other. Channel seems like the MIDI nomenclature.

File: src/PureDataNode.cpp [hunk 2]
```diff
@@ -142,6 +147,10 @@ void PureDataNode::sendMessage(const std::string &dest, const std::string &msg,
   queueTask([=](pd::PdBase &pd) { pd.sendMessage(dest, msg, list); });
 }
 
+void PureDataNode::sendNoteOn(int port, int pitch, int velocity) {
+  queueTask([=](pd::PdBase &pd) { pd.sendNoteOn(port, pitch, velocity); } );
```
Comment (notlion) [inline src/PureDataNode.cpp]: Since you've made a new `QueueItem` type, it should be used here. See how it's done in `sendFloat` etc.

**Review by notlion — CHANGES_REQUESTED** [review]
A couple comments..

**Comment by notlion** [comment]
Also I think we should have a receiver for `Note`s in the same way that we have one for bangs/floats/symbols.

File: src/PureDataNode.h (hunk 1 above)
Comment (quilime) [reply src/PureDataNode.h]: Yes -- It should be channel, not port.

**Review by quilime — COMMENTED** [review]
Renaming port to channel, and utilizing the QueueItem now.

**Comment by notlion** [comment]
Alright. After thinking about it a bit more, receiving MIDI notes will take a bit more work, since they usually are not destined for the UI thread like other messages. Let's keep this PR focused on sending MIDI notes.

I think after the port -> channel naming is fixed this can be merged.

File: src/PureDataNode.cpp [hunk 3]
```diff
@@ -142,6 +147,10 @@ void PureDataNode::sendMessage(const std::string &dest, const std::string &msg,
   queueTask([=](pd::PdBase &pd) { pd.sendMessage(dest, msg, list); });
 }
 
+void PureDataNode::sendNoteOn(int channel, int pitch, int velocity) {
+  mQueueToAudio.enqueue(QueueItem(Note { channel, pitch, velocity } ));
```
Comment (notlion) [inline src/PureDataNode.cpp]: Curly brace style should be consistent with the rest of the file.

**Comment by quilime** [comment]
Apologies in advance for the messy, sporadic commits. Definitely rebase/squash when it's done.

**Review by quilime — COMMENTED** [review]
Added a bunch more types.

**Review by notlion — APPROVED** [review]

**Pull request merged**